### ISSUE TITLE
Regex in C?

### ISSUE THREAD

**Issue by elextr** [geany/geany#402]
The pattern /something/ is highlighted as 'regex' in C, but C doesn't have regexen.

(Pointed out by cousteau on IRC, thanks)

**Comment by b4n** [comment]
I see two ways of fixing this:

1. just map `regex` style in C and C++ filetypes to something else (`error`, `identifier`, `operator` or whatever).  This won't change that it's recognized as a regex, but it wouldn't be highlighted as such.  Though, in something like `a = /0;`, the `0;` part would still have the same style as the `/`, e.g. the `0` wouldn't be highlighted as a number.
2. add a property to the C++ lexer to disable regex highlighting.  This would really disable it, but requires providing a patch to Scintilla.

PS: I don't see this as very high priority though, as I don't know of any (well, much, I know a single tricky one thanks to the preprocessor) valid C or C++ constructs where regex are incorrectly recognized.

**Comment by codebrainz** [comment]
Yeah, it really should be a lexer property in Scintilla like the other ones like "hashquoted", "backquoted", "triplequoted", etc. that nobody uses because they're completely undocumented. Maybe it already is one of those, who knows.
